routes/authentication.js

```diff
@@ -1,12 +1,46 @@
 var express = require('express');
+var ldap = require('ldapjs');
 var router = express.Router();
+var ldapUtils = require('../ldap');
+
+var client = ldap.createClient({
+    url: 'ldap://127.0.0.1:636'
+});
 
 module.exports = function () {
     //GET login
     router.get('/login', function(req, res, next) {
-        res.redirect('http://localhost:8080/authenticate/atlas');
+        res.render('login', {user: req.session.user});
     });
 
+    //Validate login credentials and redirect accordingly
+    router.post('/login', function(req, res) {
+        ldapUtils.authenticate(req.body.username, req.body.password, function(err) {
+            //If authentication fails, redirect to login
+            if(err) {
+                console.log(err);
+                res.redirect('/login');
+            }
+            else {
+                //Fetch user details
+                ldapUtils.getUser(req.body.username, function(err, user) {
+                    console.log(user);
+                    //If unable to fetch user details, redirect to login
+                    if(err) {
+                        console.log(err);
+                        res.redirect('/login');
```

Comment:
Will this pass the error back to the login page so it can be displayed?

Reply:
I'm storing the error in the session, so that it can be displayed later. What do you think of it?